prettyPrinters/providers.py

```diff
@@ -0,0 +1,114 @@
+#
+#     class SyntheticChildrenProvider:
+#         def __init__(self, SBValue, dict)
+#         def num_children(self)
+#         def get_child_index(self, str)
+#         def get_child_at_index(self, int)
+#         def update(self)
+#         def has_children(self)
+#         def get_value(self)
+#
+#
+# You can find more information and examples here:
+#   1. https://lldb.llvm.org/varformats.html
+#   2. https://lldb.llvm.org/python-reference.html
+#   3. https://lldb.llvm.org/python_reference/lldb.formatters.cpp.libcxx-pysrc.html
+#   4. https://github.com/llvm-mirror/lldb/tree/master/examples/summaries/cocoa
+################################################################################################################
+
+
+def SizeSummaryProvider(valobj, dict):
+    # type: (SBValue, dict) -> str
+    return 'size=' + str(valobj.GetNumChildren())
+
+
+class StdVecSyntheticProvider:
+    """Pretty-printer for alloc::vec::Vec<T>
+
+    struct Vec<T> { buf: RawVec<T>, len: usize }
+    struct RawVec<T> { ptr: Unique<T>, cap: usize, ... }
+    struct Unique<T: ?Sized> { pointer: NonZero<*const T>, ... }
+    struct NonZero<T>(T)
+    """
+
+    def __init__(self, valobj, dict):
+        # type: (SBValue, dict) -> StdVecSyntheticProvider
+        logger = Logger.Logger()
+        logger >> "Providing synthetic children for a Vec named " + str(valobj.GetName())
+        self.valobj = valobj
+        self.update()
+
+    def num_children(self):
+        # type: () -> int
+        return self.length
+
+    def get_child_index(self, name):
+        # type: (str) -> int
+        index = name.lstrip('[').rstrip(']')
+        if index.isdigit():
+            return int(index)
+        else:
+            return -1
+
+    def get_child_at_index(self, index):
+        # type: (int) -> SBValue
+        start = self.data_ptr.GetValueAsUnsigned()
+        address = start + index * self.element_type_size
+        element = self.data_ptr.CreateValueFromAddress("[%s]" % index, address, self.element_type)
+        return element
+
+    def update(self):
+        # type: () -> None
+        self.length = self.valobj.GetChildMemberWithName("len").GetValueAsUnsigned()  # type: int
+        self.buf = self.valobj.GetChildMemberWithName("buf")  # type: SBValue
+        self.data_ptr = self.buf.GetChildAtIndex(0).GetChildAtIndex(0).GetChildAtIndex(0)  # type: SBValue
+        self.element_type = self.data_ptr.GetType().GetPointeeType()  # type: SBType
+        self.element_type_size = self.element_type.GetByteSize()  # type: int
+
+    def has_children(self):
+        # type: () -> bool
+        return True
+
+
+def StdStringSummaryProvider(valobj, dict):
+    # type: (SBValue, dict) -> str
+    assert valobj.GetNumChildren() == 1
+
+    vec = valobj.GetChildAtIndex(0)
+    length = vec.GetNumChildren()
+    chars = [chr(vec.GetChildAtIndex(i).GetValueAsUnsigned()) for i in range(length)]
+    return "".join(chars)
```

Comment:
Now we have different reperesentation for `&str` and `String`
<img width="305" alt="screen shot 2018-10-31 at 00 05 41" src="https://user-images.githubusercontent.com/2539310/47750590-c2731480-dca0-11e8-814c-232de3e2aa87.png">
So I think we should wrap this value into quotes